akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -2230,13 +2236,13 @@ private[akka] final class StatefulMapConcat[In, Out](val f: () => In => Iterable
     override def onPush(): Unit =
       try {
         currentIterator = plainFun(grab(in)).iterator
-        pushPull()
+        pushPull(pullFirstSubElement = true)
       } catch handleException
 
     override def onUpstreamFinish(): Unit = onFinish()
 
     override def onPull(): Unit =
-      try pushPull()
+      try pushPull(pullFirstSubElement = true)
```

Comment:
Shouldn't it be `pullFirstSubElement = false` here?

Reply:
Good catch! Thanks!

Reply:
Implemented a test on Telemetry side to catch this issue.